src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1955,7 +1955,14 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
         /// In case of CREATE AS table_function() query we should use global context
         /// in storage creation because there will be no query context on server startup
         /// and because storage lifetime is bigger than query context lifetime.
-        res = table_function->execute(table_function_ast, getContext(), create.getTable(), properties.columns, /*use_global_context=*/true, /*is_insert_query=*/true);
+        auto insert_query = std::make_shared<ASTInsertQuery>();
+
+        if (create.storage && create.storage->partition_by)
+        {
+            insert_query->partition_by = create.storage->partition_by->clone();
+        }
+
+        res = table_function->execute(table_function_ast, getContext(), create.getTable(), properties.columns, /*use_global_context=*/true, insert_query);
```

Comment:
I wonder, would be a `nullptr` in case of "NO STORAGE/NO PARTITION BY" be preferable here? Otherwise we just pass the empty phone object to `table_function`.